src/pandablocks_ioc/_tables.py

```diff
@@ -679,7 +677,8 @@ def _update_scalar(self, waveform_record_name: str) -> None:
             scalar_val = waveform_data[index]
             if labels:
                 # mbbi/o records must use the numeric index
-                scalar_val = labels.index(scalar_val)
+                if isinstance(scalar_val, str):
+                    scalar_val = labels.index(scalar_val)
```

Comment:
There should be a test of some kind to cover this line. Probably doesn't have to be a full system test.